src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -316,47 +316,6 @@ public void Update(Point2D mouseCursor)
         /// </summary>
         internal class StateMachine
         {
-            #region Private Nested Class MouseClickHistory
-
-            class MouseClickHistory
-            {
-                internal int Timestamp { get; set; }
-                internal object Source { get; set; }
-                internal Point Position { get; set; }
-
-                internal MouseClickHistory(object sender, MouseButtonEventArgs e)
-                {
-                    this.Timestamp = e.Timestamp;
-                    this.Source = e.Source;
-
-                    IInputElement element = sender as IInputElement;
-                    this.Position = e.GetPosition(element);
-                }
-
-                internal static bool CheckIsDoubleClick(
-                    MouseClickHistory prevClick, MouseClickHistory curClick)
-                {
-                    if (prevClick == null || (curClick.Source != prevClick.Source))
-                        return false; // Click events did not come from same source
-
-                    int clickInterval = curClick.Timestamp - prevClick.Timestamp;
-                    if (clickInterval > System.Windows.Forms.SystemInformation.DoubleClickTime)
-                        return false; // Time difference is more than system DoubleClickTime
-
-                    double diff = Math.Abs(prevClick.Position.X - curClick.Position.X);
-                    if (diff > Configurations.DoubleClickAcceptableDistance)
-                        return false; // Click is beyond acceptable threshold.
-
-                    diff = Math.Abs(prevClick.Position.Y - curClick.Position.Y);
-                    if (diff > Configurations.DoubleClickAcceptableDistance)
-                        return false; // Click is beyond acceptable threshold.
-
-                    return true;
-                }
-            }
-
-            #endregion
-
```

Comment:
This whole logic of `MouseClickHistory` is not necessary, since `MouseButtonEventArgs.ClickCount >= 2` can accurately indicate a double/triple-click event.